src/timely-util/src/order.rs

```diff
@@ -168,6 +209,35 @@ impl<P: Eq> Interval<P> {
     }
 }
 
+impl<P: PartialOrd> Interval<P> {
+    pub fn contains(&self, other: &P) -> bool {
+        self.lower <= *other && *other <= self.upper
+    }
+}
+
+impl<P: Sequence + PartialOrd + Clone> Interval<P> {
+    /// Returns up to two intervals that contain the range before and/or after given point,
+    /// neither of which contain the point.
+    pub fn split(&self, point: &P) -> Vec<Self> {
```

Comment:
We could return a `(Option<Self>, Self, Option<Self>)` so that we don't allocate and to encode a bit more info in the type system